js/core/stdio/stdout.js

```diff
@@ -0,0 +1,31 @@
+// Copyright 2014-2015 runtime.js project authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+class stdout {
+  constructor() {
+
+  }
+
+  write(text, fg, bg) {
+    runtime.tty.print(text, 1, fg, bg);
+  }
+
+  writeln(text, fg, bg) {
```

Comment:
writeln sounds not very javascripty :)
Btw, colors are probably not the part of stdio interface, expose them as a separate `setColor/setBackground` methods and make write accept any number of arguments?
Hm, now that I think about it, I think it would be nicer to make this class a pure interface. How I see it:

```js
class Stdout {
  constructor() {
    this.onwrite = function() {}
    this.onsetcolor = function(){}
    this.onsetbackgroundcolor = function() {}
  }

  write(...args) {
   this.onwrite(args.join(' '));
  }
  
  writeLine(...args) {
   this.onwrite(args.join(' ') + '\n');
  }

  setColor(fg) {
   this.fg = fg;
  }

  setBackgroundColor(bg) {
    this.bg = bg;
  }
}

class Stdin {
  // the same idea as stdout
  constructor() {
    this.onreadchar = function() {}   
  }
 
  readChar() {
    this.onreadchar(cb);
  }
 
  readLine() {
    // read the full line here using readchar
    this.onreadchar(cb);
  }
}


// Create main stdout

var defaultStdout = new runtime.tty.Stdout();  // is this part of tty?
var defaultStdin = new runtime.tty.Stdin();

defaultStdout.onsetcolor = function() {
  // change color
};
defaultStdout.onwrite = function(data) {
  runtime.tty.print(data, fg, bg);
};
defaultStdin.onreadchar = function(cb) {
  runtime.tty.readChar(cb);
}

shell.runCommand({
  stdout: defaultStdout,
  stdin: defaultStdin,
  args: {
   key: value
  }
}, function(statusCode) {
  // completed
})

// Command

setCommand(function(env, cb) {
 env.stdout.write('Output');
});

```